### PR TITLE
Publish packaged apps

### PR DESCRIPTION
**Pull request by freaktechnik**
This allows to publish packaged apps without breaking the existing API. I sadly can't get testing to work, I always get a 403 with `{ detail: 'You do not have permission to perform this action.' }`.

EDIT: I can't find this error within the Zamboni code base, so this is probably some account permission thing on the dev env or similar.

### CODE REVIEW

**Comment by brittanydionigi** [comment]
Yes this would mean you are not authenticated properly.  I also have to see how much focus there should be on packaged apps, as it is in the roadmap for v3 to deprecate them https://wiki.mozilla.org/Gaia/Architecture_Proposal#Are_we_still_going_to_be_giving_support_for_packaged_apps.3F. Let me think on this for a bit.

**Comment by rorito** [comment]
V3 will be lucky to ship by the end of year, plus packaged apps will still ship on v2 and below. Pragmatically, they will be here for 1-2 years at least.

**Comment by brittanydionigi** [comment]
Thanks @rorito. I'll review this today @freaktechnik

File: index.js [hunk 1]
```diff
@@ -98,22 +98,31 @@ MarketplaceClient.prototype.validatePackage = function(packagePath) {
   return promise;
 };
 
-MarketplaceClient.prototype.publish = function(validatedManifestId) {
+MarketplaceClient.prototype.publish = function(validationId, type) {
+  if (type === null) {
```
Comment (brittanydionigi) [inline index.js:102]: @freaktechnik this should also check if `type === undefined`, no? Idea here is if a user is just validating a manifest URL, they can still call `publish(validationID)` leaving the type param out altogether, yes?

Comment (freaktechnik) [inline index.js:102]: I guess it'd be better to check the type of the type argument, so it is set to the default value whenever an incompatible type is used.

Comment (brittanydionigi) [inline index.js:102]: Cool I can fix this - everything else looks good

**Comment by brittanydionigi** [comment]
@freaktechnik FYI changed this logic up a bit during my fix https://github.com/mozilla/node-firefox-marketplace/commit/07942625501bc770ef2dbe63893534ad3c3ca70d

**Comment by freaktechnik** [comment]
Oh, I just noticed I forgot to document it in the readme.

**Comment by brittanydionigi** [comment]
@freaktechnik yea I just opened an issue for that. I can get to it today if you're busy otherwise feel free to send a PR

**Pull request merged**